src/main/java/com/amazon/kinesis/kafka/AmazonKinesisSinkTask.java

```diff
@@ -137,7 +148,7 @@ public void put(Collection<SinkRecord> sinkRecords) {
 			else
 				f = addUserRecord(kinesisProducer, streamName, partitionKey, usePartitionAsHashKey, sinkRecord);
 
-			Futures.addCallback(f, callback);
+			Futures.addCallback(f, callback, MoreExecutors.directExecutor());
```

Comment:
I would recommend to address this in dedicated issue and PR.

Reply:
Upgrading the Kinesis Client Library from 1.7.3 to 1.13.3 (or 1.14.0, as suggested) pulls in a newer dependency of Guava (18.0 -> 26.0). The Guava 26.0 library requires an executor be supplied to the `addCallback` method: https://guava.dev/releases/26.0-jre/api/docs/com/google/common/util/concurrent/Futures.html#addCallback-com.google.common.util.concurrent.ListenableFuture-com.google.common.util.concurrent.FutureCallback-java.util.concurrent.Executor-

Earlier versions of Guava had an overloaded `addCallback` method that allowed the executor parameter to be optional. That's no longer the case.

Unfortunately, upgrading the KCL makes this change necessary too.